feconf.py

```diff
@@ -252,6 +252,11 @@ def get_empty_ratings():
 # merely avoids checking for whether the promo bar is enabled for every Oppia
 # page visited.
 ENABLE_PROMO_BAR = True
+# Whether to enable maintenance mode on the site. For non-admins, this redirects
+# all HTTP requests to the maintenance page. This is the only check which
+# determines whether the site is in maintenance mode to avoid queries to the
+# database by non-admins.
+ENABLE_MAINTENANCE_MODE = False
```

Comment:
Nope, this is working as intended. We plan to use a separate version in GAE for this and switch versions, so this switch doesn't need to change within a version.